src/libstore/ssh.cc

```diff
@@ -141,7 +142,8 @@ Path SSHMaster::startMaster()
     logger->pause();
     Finally cleanup = [&]() { logger->resume(); };
 
-    bool wasMasterRunning = isMasterRunning();
+    if (isMasterRunning())
+        return state->socketPath;
```

Comment:
Instead of an early return, can you just put the rest of the function (until the return) instead this if?

There is no hard and fast rule, but I think like this better.

Reply:
I went with the early return cause that's what @thufschmitt recommended [in another PR's review](https://github.com/NixOS/nix/pull/8349#discussion_r1195247821). I don't mind either way, so I'll change it if you still prefer it that way.